File: src/services/MetricsService/MetricsPublisher.ts
import {Metric} from './Metric';
import {MetricsStorageWriter} from './MetricsStorage';

export interface MetricsPublisher {
  publish(metrics: Metric[]): Promise<void>;
}

export class DefaultMetricsPublisher implements MetricsPublisher {
  private metricsStorage: MetricsStorageWriter;

  constructor(metricsStorage: MetricsStorageWriter) {
    this.metricsStorage = metricsStorage;
  }

  publish(metrics: Metric[]): Promise<void> {
    return this.metricsStorage.save(metrics);
  }
}

File: android/app/src/main/java/app/covidshield/services/metrics/MetricsPublisher.kt
package app.covidshield.services.metrics

interface MetricsPublisher {
    suspend fun publish(metrics: List<Metric>)
}

class DefaultMetricsPublisher(private val metricsStorage: MetricsStorageWriter) : MetricsPublisher {

    override suspend fun publish(metrics: List<Metric>) {
        metricsStorage.save(metrics)
    }

}